File: lib/controllers/stats_controller.dart
import '../models/stats_model.dart';
import '../services/stats_service.dart';

class StatsController {
  Future<ScreeningStats?> getScreeningStats({String? from, String? to}) async {
    try {
      return await StatsService.getScreeningStats(from: from, to: to);
    } catch (e) {
      return null;
    }
  }

  Future<List<UserStat>> getUserStats({String sort = 'count'}) async {
    try {
      return await StatsService.getUserStats(sort: sort);
    } catch (e) {
      return [];
    }
  }
}

File: ai_child_backend/src/controllers/stats.controller.js
const statsModel = require('../models/stats.model');

exports.getScreeningStats = async (req, res) => {
  try {
    const { from, to } = req.query;
    const fromVal = from && from !== 'undefined' ? from : null;
    const toVal = to && to !== 'undefined' ? to : null;
    console.log('📊 STATS from:', fromVal, 'to:', toVal);
    const data = await statsModel.getScreeningStats(fromVal, toVal);
    res.json(data);
  } catch (err) {
    console.error(err);
    res.status(500).json({ message: 'Lỗi server' });
  }
};

exports.getUserStats = async (req, res) => {
  try {
    const { sort = 'count' } = req.query;
    const data = await statsModel.getUserStats(sort);
    res.json(data);
  } catch (err) {
    console.error(err);
    res.status(500).json({ message: 'Lỗi server' });
  }
};